### ISSUE TITLE
Suggest `readlink -f` for reading symbolic links

### ISSUE THREAD

**Issue by jyn514** [koalaman/shellcheck#2193]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# Obtain the symbolic link (if any)
ls -l "$shared_object" | awk '{print $NF}'
```

#### Here's what shellcheck currently says:

Line 3:
ls -l "$shared_object" | awk '{print $NF}'
^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
       ^-- SC2154: shared_object is referenced but not assigned.

#### Here's what I wanted or expected to see:

Line 3:
ls -l "$shared_object" | awk '{print $NF}'
^-- SCXXXX: Use `readlink -f` to canonicalize symbolic links

This is a bit of a "stretch" feature, I wouldn't have known what the awk bit was trying to do without reading the comment.

**Comment by typeundefined** [comment]
@jyn514 Sorry, I don't quite get. How would ShellCheck understand that the user's intention was to read namely the symlink? Are you sure that latest column will always contain a symlink destination?

**Comment by jyn514** [comment]
As far as I can tell, yes. If `$shared_object` isn't a symbolic link then it will be the original filename, which still behaves the same as `readlink`.
